in_toto/models/layout.py

```diff
@@ -87,31 +66,99 @@ class Layout(Signable):
+              A list of inspection objects describing any additional actions
+              carried out upon verification.
+
+      keys:
+              A dictionary of public keys whose private keys are used
+              to sign the metadata (link metadata) corresponding to the steps
+              of the supply chain. Each step can authorize one or more of the
+              here listed keys individually.
+
+      expires:
+              The expiration date of a layout.
+
+      readme:
+              A human readable description of the software supply chain defined
+              by the layout.
+
+    <Exceptions>
+      securesystemslib.exceptions.FormatError
+              If the instantiated layout has invalid properties, e.g. because
+              any of the assigned keyword arguments are invalid.
+
+    """
     super(Layout, self).__init__()
     self._type = "layout"
     self.steps = kwargs.get("steps", [])
     self.inspect = kwargs.get("inspect", [])
     self.keys = kwargs.get("keys", {})
     self.readme = kwargs.get("readme", "")
 
-    # Assign a default expiration (on month) if not specified
-    self.expires = kwargs.get("expires", (datetime.today() +
-        relativedelta(months=1)).strftime("%Y-%m-%dT%H:%M:%SZ"))
+    # Assign a default expiration (one month) if not expiration date is passed
+    # TODO: Is one month a sensible default? In any case, we need a valid
+    # expiration date in order for the layout object to validate.
+    # (c.f. self._validate_expires)
+    self.expires = kwargs.get("expires")
+    if not self.expires:
+      self.set_relative_expiration(months=1)
 
+    # TODO: Should we validate in the constructor or allow the user to create
+    # an invalid layout and call validate explicitly?
     self.validate()
 
+
   @property
   def type_(self):
-    """Getter for protected _type attribute. Trailing underscore used by
-    convention (pep8) to avoid conflict with Python's type keyword. """
+    """
+    <Purpose>
+      Getter for protected _type attribute.
+
+      NOTE: The trailing underscore used is by convention (pep8) to avoid
+      conflicts with Python's 'type' keyword.
+
+    <Returns>
+      The type of the metadata object, i.e. "layout" (see constructor).
+
+    """
     return self._type
 
+
   @staticmethod
   def read(data):
-    """Static method to instantiate a Layout and containing Step and Inspection
-    objects from a dictionary, e.g.:
-    {"steps": [<step data>, ...], "inspect": [<inspection data>, ...]} """
+    """
+    <Purpose>
+      Static method to instantiate a layout object from a Python dictionary,
+      e.g.: by parsing its JSON representation. The method expects any
+      contained steps and inspections to be Python dictionaries as well, and
+      tries to instantiate the corresponding objects using the step's and
+      inspection's read methods respectively.
+
+    <Arguments>
+      data:
+              A dictionary containing layout metadata.
```

Comment:
Can we add "conforming to the in-toto specification" (and probably the version too?)

Reply:
Will fix with #186.